src/Symfony/Component/Mailer/Test/TransportFactoryTestCase.php

```diff
@@ -102,16 +105,16 @@ public function testIncompleteDsnException(Dsn $dsn)
 
     protected function getDispatcher(): EventDispatcherInterface
     {
-        return $this->dispatcher ??= $this->createMock(EventDispatcherInterface::class);
+        return $this->dispatcher ??= new EventDispatcher();
     }
```

Comment:
What was the use case of those protected method ?

Cause if someone did
```
$this->getDispatcher->expect(self::once())->method('dispatch');
```
This won't work anymore (so could be called a BC break ?)

Reply:
The return type of the method is `EventDispatcherInterface`. If you want still use a mock object, you can override the method.